src/gd.h

```diff
@@ -1084,6 +1132,18 @@ BGD_DECLARE(void *) gdImageWebpPtr (gdImagePtr im, int *size);
 BGD_DECLARE(void *) gdImageWebpPtrEx (gdImagePtr im, int *size, int quantization);
 BGD_DECLARE(void) gdImageWebpCtx (gdImagePtr im, gdIOCtx * outfile, int quantization);
 
+BGD_DECLARE(void) gdImageHeifEx (gdImagePtr im, FILE * outFile, int quality, gdHeifCodec codec, gdHeifChroma chroma);
+BGD_DECLARE(void) gdImageHeif (gdImagePtr im, FILE * outFile);
+BGD_DECLARE(void *) gdImageHeifPtr (gdImagePtr im, int *size);
+BGD_DECLARE(void *) gdImageHeifPtrEx (gdImagePtr im, int *size, int quality, gdHeifCodec codec, gdHeifChroma chroma);
+BGD_DECLARE(void) gdImageHeifCtx (gdImagePtr im, gdIOCtx * outfile, int quality, gdHeifCodec codec, gdHeifChroma chroma);
+
+BGD_DECLARE(void) gdImageAvifEx (gdImagePtr im, FILE * outFile, int quality, gdHeifChroma chroma);
```

Comment:
these are creating HEIF container files using the AV1 codec right ?  not the AVIF format ?

either way, these seem redundant when you can call gdImageHeif funcs with the codec setting.

Reply:
thanks for the link.  i thought the container formats were diff.

that said, it seems like they're still redundant with the Heif variants and so should be omitted.  see the `gdImageCreateFromAvif` discussion above.

Reply:
As I've incorporated HEIF support into `gd_filename.c`, we'll require the `gdImageAvif*` methods as `gdImageHeif` will use HEVC instead of AV1.

Reply:
you can drop the Avif calls from that file.  i'm not seeing it being required here.

Reply:
Should I write an AVIF reader function inside `gd_filename.c` (or another file) then?

Reply:
let's just omit it entirely at this point.  as i noted, we can reconcile the libheif support for .avif once libavif support is merged.